scripts/credit-assignment-elo.ts

```diff
@@ -0,0 +1,161 @@
+    const id = getApplicationId(app);
+    const name = getProjectName(app) || id;
+    applicationDataMap.set(id, {
+      id,
+      name,
+      application: loadApplication(id),
+      research: loadResearch(id),
+      karmaGap: loadKarmaGap(id),
+    });
+  }
+  console.log("Finished pre-loading data.");
+
+  // Get the available review models/agents
+  const modelSpecs = await fetchModelSpecs();
+
+  // Load all reviews for all agents, associating with pre-loaded data
+  const reviewsByAgent = modelSpecs.reduce((acc, agent) => {
+    const agentName = agent?.name;
+    acc[agentName] = applications
+      .map((app) => {
+        const id = getApplicationId(app);
+        const baseData = applicationDataMap.get(id);
+        if (!baseData) return null;
+
+        const reviewData = loadReview(id, agentName);
+        if (!reviewData) return null;
+
+        return { ...baseData, review: reviewData };
+      })
+      .filter(Boolean);
+    return acc;
+  }, {} as Record<string, any[]>);
+
+  // Loop through each agent's reviews
+  for (const [agentName, agentApplicationsData] of Object.entries(reviewsByAgent)) {
+    console.log(`\n🎯 Running Elo tournament for agent: ${agentName}`);
+
+    // Initialize all ratings
+    const ratings: Record<string, number> = {};
+    for (const { id } of agentApplicationsData) {
+      ratings[id] = BASE_RATING;
+    }
+
+    // Run simulated pairwise matchups (round-robin style)
+    for (let i = 0; i < agentApplicationsData.length; i++) {
+      for (let j = i + 1; j < agentApplicationsData.length; j++) {
+        const appA = agentApplicationsData[i]!;
+        const appB = agentApplicationsData[j]!;
+
+        // Prepare data snippets for the prompt (avoid stringifying huge objects)
+        const projectAData = {
+          title: appA.name,
+          application_summary: appA.application?.project?.description,
+          research_summary: appA.research?.summary,
+          karmagap_score: appA.karmaGap?.score,
+          reviewer_comment: appA.review?.comments
+        };
+        const projectBData = {
+          title: appB.name,
+          application_summary: appB.application?.project?.description,
+          research_summary: appB.research?.summary,
+          karmagap_score: appB.karmaGap?.score,
+          reviewer_comment: appB.review?.comments
+        };
+
+        const prompt = `
+You are a grant allocator reviewing two projects. Consider all available information.
+
+Choose the one that deserves *more funding*, based on impact, clarity, roadmap, potential, and overall quality presented in the data below.
+
+You are essentially a judge in the tournament which gives a score based on each Agent's review, so it's important you strongly consider reviewer_comment along with the metric data provided.
+Respond ONLY with "A" or "B". Do NOT explain.
+
+--- Project A ---
+${JSON.stringify(projectAData, null, 2)}
+
+--- Project B ---
+${JSON.stringify(projectBData, null, 2)}
+`;
```

Comment:
Should the agent constitution be added to the prompt? Right now it's calling the same prompt 3 times but still stores them for each agent name.